pkg/storage/segments/store.go

```diff
@@ -111,20 +128,31 @@ func (s *segmentStore) Put(ctx context.Context, data io.Reader, expiration time.
 			Metadata:       metadata,
 		}
 	} else {
-		// uses overlay client to request a list of nodes
-		nodes, err := s.oc.Choose(ctx, overlay.Options{Amount: s.rs.TotalCount(), Space: 0, Excluded: nil})
+		sizedReader := SizeReader(peekReader)
+
+		// uses overlay client to request a list of nodes according to configured standards
+		nodes, err := s.oc.Choose(ctx,
+			overlay.Options{
+				Amount:       s.rs.TotalCount(),
+				Bandwidth:    sizedReader.Size(),
```

Comment:
should this be `sizedReader.Size() / s.rs.TotalCount()`? Same for `Space`, immediately below